converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/Observation/ObservationParser.cs

```diff
@@ -32,13 +32,24 @@ private static IEnumerable<Observation> ParseDicomDataset(DicomDataset dataset,
     {
         if (dataset.TryGetSequence(DicomTag.ConceptNameCodeSequence, out DicomSequence codes) && codes.Items.Count > 0)
         {
-            var code = new DicomCodeItem(codes);
+            Observation observation = null;
+            try
```

Comment:
Is there a unit test for this file?

Reply:
I left a comment already on line 48 of why we are doing this. I don't think a separate method is necessary - I am just adding a try catch. I had to refactor a little since we can't yield inside of a try catch

Reply:
Added unit test as well